src/platform/windows/win_tcpconn.c

```diff
@@ -191,7 +209,12 @@ tcp_send_cancel(nni_aio *aio, void *arg, int rv)
 	nni_mtx_lock(&c->mtx);
 	if (aio == nni_list_first(&c->send_aios)) {
 		c->send_rv = rv;
-		CancelIoEx((HANDLE) c->s, &c->send_io.olpd);
+		if (CancelIoEx((HANDLE) c->s, &c->send_io.olpd) == 0) {
+			// Some error happened. We will not have a callback. Finish AIO
```

Comment:
I concerned what the semantics here of CancelIoEx are.  If this fails, does it mean that the I/O is still pending?

It would be useful to know the error code... is it ERROR_NOT_FOUND?  If so, why?  Is this happening because cancel is racing against finish?

My biggest worry here is that we are somehow treating a failure to cancel as a statement that the I/O is not active.  If that's absolutely certainly the case, then I'm good with this.  If it isn't (because for example the operation cannot be canceled for some other reason), then this approach is unacceptably risky.

So I'm looking to better understand what the circumstances are when this fails.